### PR TITLE
[AutoDiff] Move AdjointValue out of Differentiation.cpp

### PR DESCRIPTION
**Pull request by bgogul**
Another refactoring for https://bugs.swift.org/browse/TF-993

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: include/swift/SILOptimizer/Utils/Differentiation/AdjointValue.h [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  adjVal.print(os);
+  return os;
+}
+
+} // end namespace autodiff
+} // end namespace swift
+
+#endif // SWIFT_SILOPTIMIZER_UTILS_DIFFERENTIATION_ADJOINTVALUE_H
+
+
```
Comment (rxwei) [inline include/swift/SILOptimizer/Utils/Differentiation/AdjointValue.h]: Please leave 1 trailing line. There appear to be 3 lines here.

File: include/swift/SILOptimizer/Utils/Differentiation/AdjointValue.h [hunk 2]
```diff
@@ -0,0 +1,179 @@
+//===--- AdjointValue.h - Helper class for Differentiation ----*- C++ -*---===//
```
Comment (rxwei) [inline include/swift/SILOptimizer/Utils/Differentiation/AdjointValue.h]:
```suggestion
//===--- AdjointValue.h - Helper class for differentiation ----*- C++ -*---===//
```

**Review by rxwei — APPROVED** [review]

**Comment by rxwei** [comment]
Might be a good idea to add `[AutoDiff]` to your AutoDiff PR titles -- this will make things easy to find.

**Pull request merged**